### PR TITLE
bwa: fixing build errors with gcc10+

### PR DESCRIPTION
**Pull request by snehring**
This PR resolves compilation errors in gcc 10+

### CODE REVIEW

**Comment by snehring** [comment]
The test failures don't appear to be caused by my changes.

**Comment by snehring** [comment]
Close/reopen to re-trigger tests.

File: var/spack/repos/builtin/packages/bwa/package.py [hunk 1]
```diff
@@ -38,6 +38,10 @@ def install(self, spec, prefix):
                     'Makefile')
         # use spack C compiler
         filter_file('^CC=.*', 'CC={0}'.format(spack_cc), 'Makefile')
+        # fix gcc 10+ errors
+        if self.spec.satisfies('%gcc@10:'):
+            filter_file(r'const uint8_t rle_auxtab\[8\]',
+                        'extern const uint8_t rle_auxtab[8]', 'rle.h')
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/bwa/package.py]: To get rid of the escaping, you can pass `string=True` so the first parameter is not interpreted as a regular expression.

**Review by michaelkuhn — APPROVED** [review]
Confirmed with GCC 11.

**Pull request merged**